cicecore/cicedynB/dynamics/ice_dyn_evp.F90

```diff
@@ -388,98 +714,426 @@ subroutine evp (dt)
+                                          uvelE   (:,:,iblk), vvelE    (:,:,iblk), &
+                                          uvelN   (:,:,iblk), vvelN    (:,:,iblk), &
+                                          dxN     (:,:,iblk), dyE      (:,:,iblk), &
+                                          dxT     (:,:,iblk), dyT      (:,:,iblk), &
+                                          tarear  (:,:,iblk),                      &
+                                          shear   (:,:,iblk), divu     (:,:,iblk), &
+                                          rdg_conv(:,:,iblk), rdg_shear(:,:,iblk))
+                  endif
+               enddo
+               !$OMP END PARALLEL DO
+
+               ! calls ice_haloUpdate, controls bundles and masks
+               call dyn_HaloUpdate (halo_info,        halo_info_mask, &
+                                    field_loc_center, field_type_scalar, &
+                                    zetax2T, etax2T, stresspT, stressmT)
+
+               !$OMP PARALLEL DO PRIVATE(iblk)
+               do iblk = 1, nblocks
+                  call stressC_U (nx_block,             ny_block,             &
+                                                       icellu(iblk),         &
+                                 indxui      (:,iblk), indxuj      (:,iblk), &
+                                 uvelE     (:,:,iblk), vvelE     (:,:,iblk), &
+                                 uvelN     (:,:,iblk), vvelN     (:,:,iblk), &
+                                 uvel      (:,:,iblk), vvel      (:,:,iblk), &
+                                 dxE       (:,:,iblk), dyN       (:,:,iblk), &
+                                 dxU       (:,:,iblk), dyU       (:,:,iblk), &
+                                 tarea     (:,:,iblk), uarea     (:,:,iblk), &
+                                 ratiodxN  (:,:,iblk), ratiodxNr (:,:,iblk), &
+                                 ratiodyE  (:,:,iblk), ratiodyEr (:,:,iblk), &
+                                 epm       (:,:,iblk), npm       (:,:,iblk), &
+                                 hm        (:,:,iblk),                       &
+                                 zetax2T   (:,:,iblk), etax2T    (:,:,iblk), &
+                                 strength  (:,:,iblk), shrU      (:,:,iblk), &
+                                 stress12U (:,:,iblk))                   
+               enddo
+               !$OMP END PARALLEL DO
+
+               ! calls ice_haloUpdate, controls bundles and masks
+               call dyn_HaloUpdate (halo_info         ,  halo_info_mask, &
+                                    field_loc_NEcorner,  field_type_scalar, &
+                                    stress12U)
+
+               !$OMP PARALLEL DO PRIVATE(iblk)
+               do iblk = 1, nblocks
+
+                  call div_stress (nx_block,             ny_block,             & ! E point
+                                                         icelle(iblk),         &
```

Comment:
I suggest that you inline the div_stress calculations to eliminate if conditionals

Reply:
I'm happy to do this, but it will reduce code reuse.  The conditionals are also outside the do loop, so cost is not an issue.

Reply:
Calls to div_stress are inside the subcycling, and I think this is what's inlined per the comment immediately above?

Reply:
div stress is inside the subcycling but the div_stress conditions are outside the gridcell looping (ij loop).  So the div_stress if's are called once per block, not once per gridcell.  So I think the cost of the if's isn't critical.  But I think what I will do is split div_stress into 4 methods so the compiler can inline more efficiently.  I'm not convinced it'll help performance but we can try.

Reply:
I have split div_stress into 4 subroutines, no more ifs.